src/types.ts

```diff
@@ -255,14 +255,12 @@ export type RevalidateCallback = <K extends RevalidateEvent>(
   type: K
 ) => RevalidateCallbackReturnType[K]
 
-export type StateUpdateCallback<Data = any, Error = any> = (state: {
-  data?: Data
-  error?: Error
-  isValidating?: boolean
-}) => void
+export type StateUpdateCallback<Data = any, Error = any> = (
+  state: State<Data, Error>
```

Comment:
Reuses the `State` type, which will add `isLoading` which was missing in the original type definition.

https://github.com/vercel/swr/blob/09d80c525798e85e6fd7e1dbc939b68249ad2b29/src/types.ts#L167-L172